assets/js/src/form_editor/store/actions.ts

```diff
@@ -295,3 +307,23 @@ export function* applyStylesToAllTextInputs(styles) {
     styles,
   };
 }
+
+export function createHistoryRecord() {
+  return {
+    type: 'CREATE_HISTORY_RECORD',
+  };
+}
+
+export function* historyMove(action) {
+  const editorHistoryOffset = select('mailpoet-form-editor').getEditorHistoryOffset();
+  if (editorHistoryOffset === 0) {
```

Comment:
I am not sure I understand. Why do we want to create a new action here? I mean we are at the beginning of the history, right? So I think nothing should happen when the user goes back. So I don't understand why is here the `CREATE_HISTORY_RECORD` triggered.

Reply:
This is an excellent question. I will try to explain it by example. When you make one change then you will have one history record. Right? And when you return back then you will have restored the same record that you have in history. And my question is. What do you restore when you will want to move forward? Without creating a record when you move from zero you haven't the last state. Do you agree?

Reply:
I am not sure I follow your logic, but I reckon that still needs an explanation in the code then:

> When you make one change then you will have one history record. Right?
right, I follow that logic
> And when you return back then you will have restored the same record that you have in history
I think I follow, but you mean do undo and redo, correct?
>  What do you restore when you will want to move forward?
I don't know, why do you need to restore anything, you are at the end of the queue, you are not? Why then restore something if you are at the end.

Reply:
> When you make one change then you will have one history record. Right?
You make some change in form. For example, move up with a button.
> And when you return back then you will have restored the same record that you have in history
I mean **undo** 🔙.
> What do you restore when you will want to move forward?
When you would have only one record in history, and after **undo** you want to do **redo**, you will have nothing to restore. This is the reason why I create new history record before **undo**.